frontend/class-woocommerce-shop-page.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Represents the logic to determine if the current page is a WooCommerce shop page.
+ */
+class WPSEO_WooCommerce_Shop_Page {
+
+	/**
+	 * Checks if the current page is the shop page..
+	 *
+	 * @return bool True when the current page is the shop page.
```

Comment:
I always find with boolean returns that `Whether` works very naturally. So this could become `Whether the current page is the WooCommerce shop page`.